src/services/refactors/addAsyncSuffix.ts

```diff
@@ -0,0 +1,52 @@
+/* @internal */
+namespace ts.refactor {
+    const asyncSuffixRefactor: Refactor = {
+        name: "Add Async suffix",
+        description: "Add an 'Async' suffix to async function declarations",
+        getCodeActions,
+        isApplicableForPositionOrRange
+    };
+
+    registerRefactor(asyncSuffixRefactor);
+
+    function getCodeActions(context: RefactorContext, positionOrRange: number | TextRange): CodeAction[] | undefined {
+        const { file, program } = context;
+        const tokenPos = typeof positionOrRange === "number" ? positionOrRange : positionOrRange.pos;
+        const token = getTokenAtPosition(file, tokenPos);
+
+        const functionSymbol = program.getTypeChecker().getSymbolAtLocation(token);
+        if (!(functionSymbol.flags & SymbolFlags.Function)) {
+            return undefined;
+        }
+
+        const oldNameNode = functionSymbol.valueDeclaration as NamedDeclaration;
+        if (oldNameNode === undefined) {
+            return undefined;
+        }
+        const changeTracker = textChanges.ChangeTracker.fromCodeFixContext(context);
+
+        changeTracker.replaceNode(file, oldNameNode, createIdentifier(functionSymbol.name + "Async"));
+
+        return [{
+            changes: changeTracker.getChanges(),
+            description: asyncSuffixRefactor.description
+        }];
+    }
+
+    function isApplicableForPositionOrRange(context: RefactorContext, positionOrRange: number | TextRange): boolean {

```

Comment:
Actually why not just put these on the `RefactorContext`? we put the `errorCode` on the codeFix context.